services/blockstorage/test/block_sync_test.go

```diff
@@ -0,0 +1,441 @@
+package test
+
+import (
+	"github.com/orbs-network/go-mock"
+	"github.com/orbs-network/orbs-network-go/test/builders"
+	"github.com/orbs-network/orbs-network-go/test/crypto/keys"
+	"github.com/orbs-network/orbs-spec/types/go/primitives"
+	"github.com/orbs-network/orbs-spec/types/go/protocol"
+	"github.com/orbs-network/orbs-spec/types/go/protocol/gossipmessages"
+	"github.com/orbs-network/orbs-spec/types/go/services/gossiptopics"
+	"github.com/stretchr/testify/require"
+	"testing"
+	"time"
+)
+
+func generateBlockAvailabilityRequestInput(lastCommittedBlockHeight primitives.BlockHeight, senderPublicKey primitives.Ed25519PublicKey) *gossiptopics.BlockAvailabilityRequestInput {
+	return &gossiptopics.BlockAvailabilityRequestInput{
+		Message: &gossipmessages.BlockAvailabilityRequestMessage{
+			SignedBatchRange: (&gossipmessages.BlockSyncRangeBuilder{
+				BlockType:                gossipmessages.BLOCK_TYPE_BLOCK_PAIR,
+				LastCommittedBlockHeight: lastCommittedBlockHeight,
+			}).Build(),
+			Sender: (&gossipmessages.SenderSignatureBuilder{
+				SenderPublicKey: senderPublicKey,
+			}).Build(),
+		},
+	}
+}
+
+func TestSyncSourceHandlesBlockAvailabilityRequest(t *testing.T) {
+	// FIXME user WithContext everywhere
+	//test.WithContext(func(ctx context.Context) {
+
+	driver := NewDriver()
+
+	driver.expectCommitStateDiffTimes(2)
+
+	driver.commitBlock(builders.BlockPair().WithHeight(primitives.BlockHeight(1)).WithBlockCreated(time.Now()).Build())
+	driver.commitBlock(builders.BlockPair().WithHeight(primitives.BlockHeight(2)).WithBlockCreated(time.Now()).Build())
+
+	senderKeyPair := keys.Ed25519KeyPairForTests(9)
+
+	input := generateBlockAvailabilityRequestInput(primitives.BlockHeight(0), senderKeyPair.PublicKey())
+	response := &gossiptopics.BlockAvailabilityResponseInput{
```

Comment:
Extract this call to a method to make code more readable, this is hard to follow

Reply:
Fixed.